### DISCUSSION TITLE
Disable animations

### DISCUSSION THREAD

**Discussion by mr-cheffy** [zen-browser/desktop#830] [Ideas]
### What feature would you like to see?

I want an option to disable animations. Animations sometimes can be annoying and reducing performance cost is a good thing.

### What platform are you seeing the problem on?

Linux

 
 ~ _Created automatically from issue: https://github.com/zen-browser/desktop/issues/465_

**Comment by antiquealive** [comment]
second this. on my windows, when on enable zen compact mode, aka autohide side bar, hovering mouse to show the tabbar causes 4.5% gpu spike, kind of unnecessary. maybe animation is not the culprit, not sure.

**Reply by antiquealive** [reply]
temp sol. 
`* {
    animation: none !important;
    transition: none !important;
}`

**Reply by velaar** [reply]
@antiquealive  Thank you so much for posting the update! If anyone else is looking - I did the mod via userChrome.css in chrome folder in the profile folder and it works as expected.

**Reply by ahmedelgabri** [reply]
> temp sol. `* { animation: none !important; transition: none !important; }`

Unfortunately, this breaks glance, I wonder if there is a better way without breaking glance?

**Reply by standerksen** [reply]
Does not work for workspace switching, unfortunately.

**Reply by volkmnv** [reply]
I use only ```* { animation: none !important; }``` which doesn't remove all animations but I haven't noticed any problems so far. That makes it at least a little bit better.

**Reply by BenjaminAlbertR** [reply]
Hey, I am not a programmer. Where do I enter * { animation: none !important; } as the bookmarks bar is really slow on popup. And I want to fix it.

**Comment by BitBernd** [comment]
I switched to Zen to be more productive, animations just slow me down a lot.
Please add the option to disable all animations.

**Comment by standerksen** [comment]
This should definitely be an option, I'm surprised it isn't. Another scenario where this is useful is switching workspaces using hotkeys. Currently we have to wait for the animation to finish before we can switch to the next. This is really annoying when you quickly want to switch two or three workspaces over.

**Comment by 11Spades** [comment]
I'd also love to see this feature. I value the responsiveness of my software very highly, and while Zen's animations are nice, I'd much prefer to have as close to an instantaneous response to my inputs as possible.

**Comment by ajpocus** [comment]
Agreed, I would also like to see this feature and I'll add that the media player animation in particular is distracting -- my eye keeps catching it in my peripheral vision while I'm trying to browse.

**Comment by phannawich** [comment]
I also would love to see this feature, I don't like the animation when switching workspace. If I can disable it, would be nice.

**Comment by Sabreclaw** [comment]
Thank you for the amazing browser itself! I would definitely love to see this feature, if possible even contribute in it.

**Comment by Dalai-** [comment]
Surprised to find there isn't an option for this. I was looking in about:config for one, but the options there seem unrelated.

**Comment by yunidh** [comment]
bumping this issue cause still no update

**Comment by ckx** [comment]
I am loving most everything about Zen, but after combing through the settings & about:config I can't figure out how to disable animations for stuff like:

* "Pop-up" animations, e.g. when pressing "..." for the menu, or opening ublock origin's menu.
* Bouncy animation when selecting a tab
* Workspace switching

These adversely affect responsiveness and add unnecessary distraction, both huge factors for me, so I may have to hold-off on fully transitioning to Zen for the moment.

**Comment by tapafon** [comment]
Original Firefox does respect "Accessibility->Visual effects->Animation effects" setting in Windows or similar setting in KDE on Linux. Zen only does it partially (new animations introduced by Zen don't respect that setting).
I prefer having little or no animations system-wide. Zen ruins it (especially bouncy animation when selecting a tab, especially when working via RDP/VNC).

**Comment by p8128n** [comment]
Any update on a way to disable these animations. I would think this is a very simple code change. I frequently jump between multiple work spaces in succession and this really slows things down to the point of being a productivity blocker. I'm holding off on using Zen full time due to this feature alone.

**Comment by zuble** [comment]
```
about:config -> "zen.view"

zen.view.compact.toolbar-flash-popup false   # <--- no sidebar show when tab_switch@compact
zen.view.compact.toolbar-hide-after-hover.duration 200 
zen.view.compact.animate-sidebar false
```

**Reply by Clam-** [reply]
?_?

![zencompact](https://github.com/user-attachments/assets/04c14197-8722-4e83-8861-627f9a05d112)

**Reply by zuble** [reply]
yy , got excited for not having delay/sidebar when switching tabs 

there's still animations all over :(

**Comment by brimwats** [comment]
The URLbar animaltions are noticably laggy and distracting on my computer and make me kind of nauseous.

**Comment by ysomad** [comment]
Would like to see the feature to disable every animation possible. Would be a game changer for a lot of people

**Comment by idea90** [comment]
where is update?

**Comment by mr-cheffy** [comment]
Not exactly sure what animations there are in zen to be disabled? Specially since some people mention the "urlbar"?

**Reply by standerksen** [reply]
I have no issues with the URL bar, but it's mostly the workspace switching animation that is a problem for me. If you want to quickly move 2 or more workspaces over you have to wait for every animation to finish, which can get quite annoying if you do it often.

**Reply by Clam-** [reply]
> Not exactly sure what animations there are in zen to be disabled? Specially since some people mention the "urlbar"?

I posted a gif earlier:
![zencompact](https://github.com/user-attachments/assets/04c14197-8722-4e83-8861-627f9a05d112)

Do you not see that behavior on your end? When the sidebar "slides out" from the side. That animation is one of them.

**Reply by mr-cheffy** [reply]
These are very minimal transitions that act upon an explicit action. I dont think reduced motion should be respected in both of the examples, because removing them would eliminate essential feedback and make the interaction feel abrupt, less intuitive, and potentially harder to understand for the user

**Reply by standerksen** [reply]
I think if you got out of your way to manually disable these animations (with a setting for instance) you know what you are signing up for.

I understand it looks like it is a minimal change, but just think about how often someone would switch workspace. These add up.

**Reply by mr-cheffy** [reply]
Maybe instead of straight up killing the animation, we should rather handle the space switching animations.

Right now, they are blocked until the animation finishes. Maybe we should finish the previous animation instantly if its still running once the user tries to change the space again

**Reply by ckx** [reply]
> These are very minimal transitions ...

This is relative. What is minimal to you may be a deal breaker for another. One of the core tenets of Zen is to stay focused with minimal distractions, but these animations break flow both due to being distracting in nature, and slowing me down.

Although I love the idea and most of the execution, the browser does not fulfill its intended objective for me in the current state.

**Reply by Clam-** [reply]
> I dont think reduced motion should be respected in both of the examples, because removing them would eliminate essential feedback and make the interaction feel abrupt, less intuitive, and potentially harder to understand for the user

Okay, this is a fair response if we are talking about "default behaviour". If you want default behaviour to give the user transitional extended feedback, that's fine.
However I don't think anyone in this thread is talking about changing default behaviour. This request is to have an option to turn off that default behaviour for users that find excessive motion/delay bothersome.
Additionally, what's the point of the option "animate-sidebar" if turning it off doesn't turn it off ?_?

**Reply by standerksen** [reply]
> Maybe instead of straight up killing the animation, we should rather handle the space switching animations.
> 
> Right now, they are blocked until the animation finishes. Maybe we should finish the previous animation instantly if its still running once the user tries to change the space again

That would be a amazing and fix my issue.

**Reply by mr-cheffy** [reply]
>  Additionally, what's the point of the option "animate-sidebar" if turning it off doesn't turn it off ?_?

I think thats used for when toggling only, not hovering.

The issue is that reduced motion is often enabled very easily by environmental factors such as low battery mode and also commonly enabled by mistake by users without noticing, thats why I believe there are certain transitions in zen that shouldn't respect this preference. For example, if the space switching animation where to be instant, the user could experience a complete UI change without having any sort of transitional feedback along with it. This creates more harm than help imho because it creates confusion.

That said, they are also really quick and seeming-less transitions, they aren't 5 second scaling animations. AFAIK reduced motion exists to minimize **non-essential** animations, to help (for example) people with vestibular motion disorders feel less discomfort. I believe compact mode and space switching animations are essential.

I'll send here soon a PR here that handles some of the concerns from @standerksen and @ckx

**Reply by Clam-** [reply]
> > Additionally, what's the point of the option "animate-sidebar" if turning it off doesn't turn it off ?_?
> 
> I think thats used for when toggling only, not hovering.

Ohhhh, that makes much more sense now. Using the hotkey doesn't show the animation. Great. I can deal with that. Thank you for clarifying.

**Reply by mr-cheffy** [reply]
Bit of an epilepsy warning: 

Left completing the animation if one is already running vs current blocking implementation (right) @standerksen  

https://github.com/user-attachments/assets/6f5233a5-5e49-460f-b830-fee743fcbf1b

https://github.com/zen-browser/desktop/pull/12980

**Reply by e5150** [reply]
> These are very minimal transitions that act upon an explicit action. I dont think reduced motion should be respected in both of the examples, because removing them would eliminate essential feedback and make the interaction feel abrupt, less intuitive, and potentially harder to understand for the user

When I give an explicit instruction I just expect the program to perform the requested action, without telling me what it's doing. Any fading, sliding or other visual transition grabs my attention and just distracts me from whatever task I am doing. To me it's essential to not have this kind of feedback.

**Reply by mr-cheffy** [reply]
What animation are we talking about right now? Which one is distracting you?

**Reply by brimwats** [reply]
Frankly, we have to go through so many steps to disable animations at an OS level and they're incredibly disabling, they always break assistive and disability devices and it's just basic a11ly stuff that animations should be disable-able; I use zen because its the best but it the URLbar and the workspace animations absolutely choke my computer for minutes, the CPU spikes to 99% because the assistive tech that turn animations and the zen browser are fighting each other. Just let us turn off animations; use an advanced flag if you need, but it really sucks for disabled people to come here and ask nicely for an access thing and then be told that we're wrong and we don't know what we need.

**Reply by mr-cheffy** [reply]
Everyone keeps mentioning URLbar animations, the urlbar desn't have any animations?

**Reply by borisv** [reply]
What ever you call it: For me it is the URLbar popping up when hitting Ctrl-T - when you then use the arrow down to step to an entry it pulsates the first two entries, this is the kind of rubbish and completely useless crap that gets in the way of being actually zen. Believe me, it is alright and already does its job perfectly when it doesn't bling-bling and flash, change its color in a thousand ways and avoids adding crap like modulating shadows underneath.

**Reply by mr-cheffy** [reply]
https://github.com/user-attachments/assets/9d7197b7-194a-4e13-8928-7ec0155833e9

like this? Im not getting anything

**Reply by borisv** [reply]
Exactly. Mine started to pulsate after an update, a while ago.

**Reply by mr-cheffy** [reply]
Could you please show a video?

**Reply by borisv** [reply]
Irritating, I do not experience this on a windows 11 machine, while having it on fedora cosmic.

**Reply by borisv** [reply]
Cosmic's Wayland implementation makes screen recording almost impossible at this time.

**Reply by Clam-** [reply]
It's okay I can show you the pulsating as mine does it (macOS):

https://github.com/user-attachments/assets/9534b31e-2d50-46db-9713-e000ea2948e6

**Reply by mr-cheffy** [reply]
Oh I see, could you please open up an issue so it can be triaged? Something along the lines of "Urlbar switch animation doesn't respect reduce motion"

**Reply by Clam-** [reply]
Just to make sure I'm capturing it fairly: how do I make sure I am in "reduce motion" state?

**Reply by mr-cheffy** [reply]
Zen respects system settings, so setting "use reduce motion" (depends on the OS) should be enough

**Reply by Clam-** [reply]
Okay, I had to restart the browser after changing system setting for it to take effect. It now doesn't "pulse", but it still has glow animations. I'm assuming this doesn't quality as 'motion' for the reduce functionality. Here's what it looks like with that system setting: 

https://github.com/user-attachments/assets/95757624-a712-4303-911d-f3e4b00e1592

**Reply by mr-cheffy** [reply]
@brimwats `1.19.8b` should fix some of your concerns, specially with the urlbar animation

**Comment by ckx** [comment]
This issue has still been blocking me from switching to Zen. I follow it closely and check the browser now and then. What I have here is by no means comprehensive—I just reinstalled Zen to record a couple more in addition to what's already posted here.

### Workspace animations

I did not even have to use workspaces to see the animation as it is on the Welcome screen. But bonus for a hover-over animation on the "Space" text and a fade in on the menu.

[Screencast_20260329_052301.webm](https://github.com/user-attachments/assets/425a6a26-1c31-464f-a8c0-2e4193acc3ea)

### Menu slide-ins

I feel like there are a couple of these around.

[Screencast_20260329_052125.webm](https://github.com/user-attachments/assets/e706dc23-8a68-4ace-a26b-f04f8bd83c0f)

### New tab animation

Animation when creating a new tab:

[Screencast_20260329_051823.webm](https://github.com/user-attachments/assets/c05a3321-2ff0-48b8-b74b-5cb4d2a5f245)

Compare with vanilla firefox:

[Screencast_20260329_053142.webm](https://github.com/user-attachments/assets/d94c2d78-87f2-4061-82d7-c35d3d5df1e0)

### Tab selection animation

I recall this one being bouncier in the past, but it still does an animation when selecting tabs that I would prefer to not be there.

[Screencast_20260329_051703.webm](https://github.com/user-attachments/assets/c114f42e-9390-48ee-b931-383c2a5fb1b0)

**Comment by borisv** [comment]
Please, don't make a good thing worse, let us turn off animations! I do not need and do not want flashing and animating effects. If you implement stuff, allow for turning it off - regardless the platform.

**Reply by untitled500** [reply]
Yes, implement animations if you want, but only if you allow them to be turned off.

**Comment by vunhatchuong** [comment]
To clarify, I opened this issue because Zen is a browser that promotes the idea “Browsers should be tools that help you get things done, not distractions that keep you from your work.” Animations would be a non-issue if it were something like Opera GX - the gaming browserTM.

**Comment by bungadrum** [comment]
Can you disable the animation when opening a webpage in a floating window? Make it open and close instantly without any animations or effects.

https://github.com/user-attachments/assets/087c9a61-5e03-490c-bb54-f09db49ca0a2

**Reply by mr-cheffy** [reply]
Could you please open up an issue about it? So it can be tracked. "Glance doesn't respect reduce motion"

**Comment by shalva97** [comment]
Life is too short for animations

**Comment by RooneyMcNibNug** [comment]
Came here to request this in particular for the audio player animations. They are kind of eye-catching in a bad way, and it would be nice to toggle these off for people like myself who find it distracting:

<img width="222" height="69" alt="image" src="https://github.com/user-attachments/assets/717dfc8c-bad1-4a2e-8a2c-d127e75772a6" />

**Comment by shalva97** [comment]
Life is also short to mark my comments as low quality
